Formula/fish.rb

```diff
@@ -22,9 +22,16 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head? || build.devel?
+    args = %W[
+      --prefix=#{prefix}
+      --with-extra-functionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_functions.d
```

Comment:
`#{HOMEBREW_PREFIX}/share/fish` on each line should be `#{pkgshare}` instead; it's failing because Homebrew doesn't allow writing outside the formula's Cellar during installation.